flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliChangelogResultView.java

```diff
@@ -124,18 +125,11 @@ protected void refresh() {
                 stopRetrieval(false);
                 break;
             default:
-                List<Tuple2<Boolean, Row>> changes = result.getPayload();
+                List<Row> changes = result.getPayload();
 
-                for (Tuple2<Boolean, Row> change : changes) {
+                for (Row change : changes) {
                     // convert row
-                    final String[] changeRow = new String[change.f1.getArity() + 1];
-                    final String[] row = PrintUtils.rowToString(change.f1);
-                    System.arraycopy(row, 0, changeRow, 1, row.length);
-                    if (change.f0) {
-                        changeRow[0] = "+";
-                    } else {
-                        changeRow[0] = "-";
-                    }
+                    final String[] row = PrintUtils.rowToString(change, NULL_COLUMN, true);
```

Comment:
It seems that `PrintUtils.rowToString` doesn't correctly stringify values of TIMESTAMP type. I created FLINK-21456 to track this effort.

Reply:
Sure. I will take it.